spec/requests/api/v1/customers_spec.rb

```diff
@@ -81,6 +81,17 @@
         get "/api/v1/customers", params: { page: "2", per_page: "1" }
         expect(json_response_ids).to eq [customer2.id.to_s]
       end
+
+      it "renders beyond the available pages" do
+        get "/api/v1/customers", params: { page: "2" }
+        expect(json_response_ids).to eq []
+      end
+
+      it "informs about invalid pages" do
+        get "/api/v1/customers", params: { page: "0" }
+        expect(json_response_ids).to eq nil
+        expect(json_error_detail).to eq 'expected :page >= 1; got "0"'
```

Comment:
This seems like a weird case...

We could just avoid it entirely by forcing the value to always be positive at the point we're doing the pagination, eg:
```rb
params[:per_page].to_i.clamp(1..)
```

What do you think?

Reply:
Tempting. Much simpler. But I think that we should give people direct feedback about invalid parameters. It may have been a typo and we don't want to let our sever work if not needed. And someone may have intended to get page 10 and then got page 1 instead, may not notice.